opm/grid/cpgrid/CpGridData.hpp

```diff
@@ -536,11 +543,24 @@ class CpGridData
         return level_;
     }
     /// Add doc/or remove method and replace it with better approach
-    std::vector<std::shared_ptr<Dune::cpgrid::CpGridData>> levelData() const
+    const std::vector<std::shared_ptr<Dune::cpgrid::CpGridData>>& levelData() const
     {
+        if (level_data_ptr_->empty()) {
+            OPM_THROW(std::runtime_error, "Level data has not been initialized\n");
```

Comment:
Isn't this more a `logic_error` as it actually should never happen unless there is a bug in the implementation?

Reply:
You're right. Wrong error type. I'll replace it by logic_error